File: sdk/nodejs/src/resources/affiliate.js
"use strict";

const { BaseResource } = require("./base");

class AffiliateResource extends BaseResource {
  async getSummary() {
    return this._http.get("$/api/affiliates/summary");
  }

  async listCampaigns() {
    return this._http.get("$/api/affiliates/campaigns");
  }

  async listCommissions(params = {}) {
    return this._http.get("$/api/affiliates/commissions");
  }

  async listPayouts() {
    return this._http.get("$/api/affiliates/payouts");
  }

  async listVouchers() {
    return this._http.get("$/api/affiliates/vouchers");
  }

  async listCommissionPlans() {
    return this._http.get("$/api/affiliates/commissionplans");
  }

}

module.exports = { AffiliateResource };

File: sdk/csharp/Resources/Affiliate.cs
namespace HiTechCloud.SDK.Resources;

public class Affiliate : BaseResource
{
    public Affiliate(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> SummaryAsync()
    {
        return await Http.GetAsync("/api/affiliates/summary");
    }

    public async Task<Dictionary<string, object?>> CampaignsAsync()
    {
        return await Http.GetAsync("/api/affiliates/campaigns");
    }

    public async Task<Dictionary<string, object?>> CommissionsAsync()
    {
        return await Http.GetAsync("/api/affiliates/commissions");
    }

    public async Task<Dictionary<string, object?>> PayoutsAsync()
    {
        return await Http.GetAsync("/api/affiliates/payouts");
    }

    public async Task<Dictionary<string, object?>> VouchersAsync()
    {
        return await Http.GetAsync("/api/affiliates/vouchers");
    }

    public async Task<Dictionary<string, object?>> CommissionPlansAsync()
    {
        return await Http.GetAsync("/api/affiliates/commissionplans");
    }
}
